control-base/src/main/java/fi/nls/oskari/control/layer/SaveLayerHandler.java

```diff
@@ -184,6 +185,12 @@ private SaveResult saveLayer(final ActionParameters params) throws ActionExcepti
 
                 ml.setUpdated(new Date(System.currentTimeMillis()));
                 mapLayerService.update(ml);
+
+                int[] groupIds = getMaplayerGroupIds(params.getHttpParam(PARAM_MAPLAYER_GROUPS, "-1"));
+                layerGroupLinkService.deleteLinksByLayerId(ml.getId());
```

Comment:
PARAM_MAPLAYER_GROUPS is not always sent by the frontend (when editing a layer and NOT opening the groups dialog) so this needs a check if the parameter was present.